includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -409,31 +409,41 @@ public function __construct(
 		if ( $this->is_saved_cards_enabled() ) {
 			array_push( $this->supports, 'tokenization', 'add_payment_method' );
 		}
+	}
 
-		add_action( 'woocommerce_update_options_payment_gateways_' . $this->id, [ $this, 'process_admin_options' ] );
-		add_action( 'admin_notices', [ $this, 'display_errors' ], 9999 );
-		add_action( 'woocommerce_order_actions', [ $this, 'add_order_actions' ] );
+	/**
+	 * Initializes this class's WP hooks.
+	 *
+	 * @return void
+	 */
+	public function init_hooks() {
+		add_action( 'woocommerce_update_order', [ $this, 'schedule_order_tracking' ], 10, 2 );
```

Comment:
I'm just thinking out aloud here, but I was wondering whether this hook should also be included amongst those below only by attached by the main gateway. Currently, the `schedule_order_tracking` function has no gateway specific logic...but I think this is actually a bug.

[This line](https://github.com/Automattic/woocommerce-payments/blob/add/init-hooks/includes/class-wc-payment-gateway-wcpay.php#L2888) will exit out of the hook, if the order's payment method is not equal to the gateway ID, which is fine now...because the only gateway that can be used with subscriptions is the card gateway. However, when SEPA and other PMs are enabled for recurring usage, we will need to add logic to check for their gateway IDs as well.

Basically, I think this hook is currently in the right place and you don't need to change anything, but when we sort out SEPA and subscriptions, we will need to return here and adapt the `schedule_order_tracking` function to work for all purposes. Anyways, apologies for leaving another comment that can be ignored, but this is more so that I remember to return to this in the future lol.